pages/signin.js

```diff
@@ -73,6 +78,9 @@ class SigninPage extends React.Component {
         this.setState({ error: 'Token rejected' });
       }
     }
+    if (this.state.TOTPCode && !oldState.TOTPCode) {
+      await this.props.login(this.props.token, true, this.state.TOTPCode);
```

Comment:
I would plug that directly when submitting the form, instead of waiting for a change of state.